addon-resizer/nanny/kubernetes_client.go

```diff
@@ -18,23 +18,44 @@ package nanny
 
 import (
 	"fmt"
+	"io"
+	"strings"
 
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
 	"k8s.io/client-go/kubernetes"
 	scheme "k8s.io/client-go/kubernetes/scheme"
 )
 
+const (
+	// objectCountMetricName is the metric to be used to get number of nodes
+	objectCountMetricName = "etcd_object_counts"
+	// nodeResourceName is the label value for Nodes in objectCountMetricName metric.
+	nodeResourceName = "nodes"
+	// resourceLabel is the label name for resource.
+	resourceLabel = "resource"
+)
+
 type kubernetesClient struct {
 	namespace  string
 	deployment string
 	pod        string
 	container  string
 	clientset  *kubernetes.Clientset
+	useMetrics bool
 }
 
 func (k *kubernetesClient) CountNodes() (uint64, error) {
+	if k.useMetrics {
+		return k.countNodesThroughMetrics()
+	}
+	return k.countNodesThroughAPI()
+}
+
+func (k *kubernetesClient) countNodesThroughAPI() (uint64, error) {
```

Comment:
Can you leave a one line comment on the two different methods of getting the node count here or in the CountNodes() method?

Reply:
done